test/unit/puppetlabs/services/jruby/jruby_core_test.clj

```diff
@@ -12,7 +12,7 @@
 (def min-config
   {:product
    {:name "puppet-server", :update-server-url "http://localhost:11111"},
-   :jruby-puppet
+   :jruby
```

Comment:
Assuming we move forward with my suggestion above, about preferring a `:jruby` section but falling back to `:jruby-puppet` if it doesn't exist, then I think you should leave all of your test changes as they are in this PR (i.e., test the new setting by default).  And then maybe we just add one or two unit-ish tests that validate that we will fall back to `:jruby-puppet` if `:jruby` doesn't exist?